butane_cli/src/lib.rs

```diff
@@ -231,6 +237,55 @@ pub fn load_connspec(base_dir: &PathBuf) -> Result<db::ConnectionSpec> {
     }
 }
 
+/// Load [`db::Backend`] from the names provided, or when `backend_names` is empty,
+/// fallback to the backends selected in the latest migration, or when there are no migrations,
+/// fallback to the backend named in the connection.
+pub fn load_backends(base_dir: &PathBuf, backend_names: &Vec<String>) -> Vec<Box<dyn db::Backend>> {
+    let mut backends: Vec<Box<dyn db::Backend>> = vec![];
+    if backend_names.is_empty() {
+        if let Ok(spec) = db::ConnectionSpec::load(base_dir) {
+            // Try to use the same backends as the latest migration.
+            if let Ok(ms) = get_migrations(base_dir) {
+                if let Some(latest_migration) = ms.latest() {
+                    if let Ok(backends) = latest_migration.sql_backends() {
+                        return load_backends(base_dir, &backends);
+                    }
+                }
+            }
+
+            // Finally use the backend used during `init`.
+            backends.push(spec.get_backend().unwrap());
+        } else {
+            eprintln!(
+                "No Butane connection info found. Either run `butane init` or use --backend."
+            );
+            std::process::exit(1);
+        }
+    } else {
+        // Verify named backends were in the latest migration.
+        if let Ok(ms) = get_migrations(base_dir) {
+            if let Some(latest_migration) = ms.latest() {
+                if let Ok(migration_backends) = latest_migration.sql_backends() {
+                    for backend_name in backend_names {
+                        if !migration_backends.contains(backend_name) {
+                            eprintln!("Adding new backend {backend_name} is not supported.  Use collapse to use a new backend.");
```

Comment:
"Use collapse to use a new backend." doesnt make sense yet , as collapse doesnt allow `--backend` args.